servers/src/main/java/tachyon/worker/block/evictor/LRFUEvictor.java

```diff
@@ -0,0 +1,345 @@
+    mManagerView = view;
+    mTachyonConf = new TachyonConf();
+    mStepFactor = mTachyonConf
+        .getDouble(Constants.WORKER_EVICT_STRATEGY_LRFU_STEP_FACTOR, DEFAULT_STEP_FACTOR);
+    mAttenuationFactor = mTachyonConf
+        .getDouble(Constants.WORKER_EVICT_STRATEGY_LRFU_ATTENUATION_FACTOR, 
+            DEFAULT_ATTENUATION_FACTOR);
+
+    // Preloading blocks
+    for (StorageTierView tier : mManagerView.getTierViews()) {
+      for (StorageDirView dir : tier.getDirViews()) {
+        for (BlockMeta block : dir.getEvictableBlocks()) {
+          mBlockIdToLastAccessTime.put(block.getBlockId(), 0L);
+          mBlockIdToCRFValue.put(block.getBlockId(), 0.0);
+        }
+      }
+    }
+  }
+
+  /**
+   * @return a StorageDirView in the range of location that already has availableBytes larger than
+   *         bytesToBeAvailable, otherwise null
+   */
+  private StorageDirView selectDirWithRequestedSpace(long bytesToBeAvailable,
+      BlockStoreLocation location) {
+    if (location.equals(BlockStoreLocation.anyTier())) {
+      for (StorageTierView tierView : mManagerView.getTierViews()) {
+        for (StorageDirView dirView : tierView.getDirViews()) {
+          if (dirView.getAvailableBytes() >= bytesToBeAvailable) {
+            return dirView;
+          }
+        }
+      }
+      return null;
+    }
+
+    int tierAlias = location.tierAlias();
+    StorageTierView tierView = mManagerView.getTierView(tierAlias);
+    if (location.equals(BlockStoreLocation.anyDirInTier(tierAlias))) {
+      for (StorageDirView dirView : tierView.getDirViews()) {
+        if (dirView.getAvailableBytes() >= bytesToBeAvailable) {
+          return dirView;
+        }
+      }
+      return null;
+    }
+
+    StorageDirView dirView = tierView.getDirView(location.dir());
+    return (dirView.getAvailableBytes() >= bytesToBeAvailable) ? dirView : null;
+  }
+
+  /**
+   * A recursive implementation of cascading LRFU eviction.
+   *
+   * It will try to free space in next tier view to transfer blocks there, if the next tier view
+   * does not have enough free space to hold the blocks, the next next tier view will be tried and
+   * so on until the bottom tier is reached, if blocks can not even be transferred to the bottom
+   * tier, they will be evicted, otherwise, only blocks to be freed in the bottom tier will be
+   * evicted.
+   *
+   * this method is only used in {@link #freeSpaceWithView}
+   *
+   * @param bytesToBeAvailable bytes to be available after eviction
+   * @param location target location to evict blocks from
+   * @param plan the plan to be recursively updated, is empty when first called in
+   *        {@link #freeSpaceWithView}
+   * @param sortedCRF sorted CRF of all blocks in ascending order
+   * @return the first StorageDirView in the range of location to evict/move bytes from, or null if
+   *         there is no plan
+   */
+  protected StorageDirView cascadingEvict(long bytesToBeAvailable, BlockStoreLocation location,
```

Comment:
Same here, is it identical to LRUEvictor? if yes, we shall really try to reuse code instead of duplicating code.